packages/react-native/React/Fabric/Mounting/ComponentViews/Image/RCTImageComponentView.mm

```diff
@@ -168,24 +168,41 @@ - (void)didReceiveImage:(UIImage *)image metadata:(id)metadata fromObserver:(con
   }
 }
 
-- (void)didReceiveProgress:(float)progress fromObserver:(const void *)observer
+- (void)didReceiveProgress:(float)progress loaded:(int64_t)loaded total:(int64_t)total  fromObserver:(const void *)observer
 {
   if (!_eventEmitter) {
     return;
   }
 
-  static_cast<const ImageEventEmitter &>(*_eventEmitter).onProgress(progress);
+  static_cast<const ImageEventEmitter &>(*_eventEmitter).onProgress(progress, loaded, total);
 }
 
-- (void)didReceiveFailureFromObserver:(const void *)observer
+- (void)didReceiveFailure:(NSError *)error fromObserver:(const void *)observer
 {
   _imageView.image = nil;
 
   if (!_eventEmitter) {
     return;
   }
 
-  static_cast<const ImageEventEmitter &>(*_eventEmitter).onError();
+  ImageErrorInfo info;
+
+  if (error) {
+    info.error = [error.localizedDescription UTF8String];
```

Comment:
The error string here is only alive while the `NSError` is. We should wrap in `std::string` to make copy.

https://developer.apple.com/documentation/foundation/nsstring/1411189-utf8string?language=objc